src/libfetchers/fetch-settings.hh

```diff
@@ -94,6 +93,22 @@ struct FetchSettings : public Config
           empty, the summary is generated based on the action performed.
         )",
         {}, true, Xp::Flakes};
+
+    Setting<bool> trustGitHub{
+        this, true, "trust-github",
+        R"(
+          If enabled (the default), Nix will consider trees from
+          GitHub and similar providers to be locked if a Git revision
+          is specified,
+          e.g. `github:NixOS/patchelf/7c2f768bf9601268a4e71c2ebe91e2011918a70f`.
+          This requires Nix to trust that the provider will return the
+          correct contents for the specified Git revision.
+
+          If disabled, GitHub trees are only considered locked if a
+          `narHash` attribute is specified,
+          e.g. `github:NixOS/patchelf/7c2f768bf9601268a4e71c2ebe91e2011918a70f?narHash=sha256-PPXqKY2hJng4DBVE0I4xshv/vGLUskL7jl53roB8UdU%3D`.
```

Comment:
```suggestion
          e.g. `github:NixOS/patchelf/7c2f768bf9601268a4e71c2ebe91e2011918a70f?narHash=sha256-PPXqKY2hJng4DBVE0I4xshv/vGLUskL7jl53roB8UdU%3D`.
          This `narHash` is not guaranteed to be required in future versions.
```

Not a promise, which we should be careful about, especially in reference docs, but some expectation management is in order.

Reply:
I don't think we have to document that requirements could be loosened in the future, since that wouldn't break anything.